### ISSUE TITLE
JOSS REVIEW: Missing dependency (pypulseq==0.0.3)

### ISSUE THREAD

**Issue by mathieuboudreau** [imr-framework/virtual-scanner#24]
This is for openjournals/joss-reviews#1637

Per your [installation instructions](https://github.com/imr-framework/virtual-scanner/blame/12b6a47e2fce96588adf51964474bb30540bb97d/README.md#L13), I tried to installed all your requirements (using `pip install -e .`), however I encountered a missing dependency:

```
Collecting pypulseq==0.0.3 (from virtual-scanner==1.0.21)
  ERROR: Could not find a version that satisfies the requirement pypulseq==0.0.3 (from virtual-scanner==1.0.21) (from versions: 1.0.0, 1.2.1)
ERROR: No matching distribution found for pypulseq==0.0.3 (from virtual-scanner==1.0.21)
```

It appears that this version of pypulseq is not available on pypi: https://pypi.org/project/pypulseq/#history

Can you update your requirements file, or resolve this in another way?

**Comment by mathieuboudreau** [comment]
Digging into this a bit more, I see that your requirements.txt file in your master correctly suggests version 1.0.0 of pypulseq, so it's not quite clear to me why it's trying to pull version 0.0.3. I'll look into this and report back.

**Comment by mathieuboudreau** [comment]
The conflict is in your setup.py file, which doesn't call your requirements.txt file.

https://github.com/imr-framework/virtual-scanner/blob/12b6a47e2fce96588adf51964474bb30540bb97d/setup.py#L42

**Comment by mathieuboudreau** [comment]
There is a mismatch between your setup.py file and your requirements.txt file. At the very least, this should be cleaned up so they match, but ideally you wouldn't duplicate these.

You could either suggest that the user installs the requirements.txt file using `pip install -r requirements.txt`, or create a setup.py like [this one](https://github.com/neuropoly/axondeepseg/blob/master/setup.py), which calls the requirements.txt file on [line 36](https://github.com/neuropoly/axondeepseg/blob/ebf74c1a928ef946304d821e1e5abf0b692fd440/setup.py) using [lines 14-17](https://github.com/neuropoly/axondeepseg/blob/ebf74c1a928ef946304d821e1e5abf0b692fd440/setup.py#L14-L17).

**Comment by mathieuboudreau** [comment]
Note that this bug is present while installing using pypi  using `pip install virtual-scanner`, so I would recommend using the setup.py solution I recommend above if you plan on hosting on pypi.

**Comment by sravan953** [comment]
> There is a mismatch between your setup.py file and your requirements.txt file. At the very least, this should be cleaned up so they match, but ideally you wouldn't duplicate these.
> 
> You could either suggest that the user installs the requirements.txt file using `pip install -r requirements.txt`, or create a setup.py file structured like [this one](https://github.com/neuropoly/axondeepseg/blob/master/setup.py), which calls the requirements.txt file on [line 36](https://github.com/neuropoly/axondeepseg/blob/ebf74c1a928ef946304d821e1e5abf0b692fd440/setup.py) using [lines 14-17](https://github.com/neuropoly/axondeepseg/blob/ebf74c1a928ef946304d821e1e5abf0b692fd440/setup.py#L14-L17).

Was not aware of this, thank you very much! :)
Fixed in  f3718aca6c57dc634df2cc047af5d80fc711d90f.
